src/backend/api/Fusion.Resources.Domain/Queries/GetDepartments.cs

```diff
@@ -35,24 +49,102 @@ public GetDepartments StartsWith(string department)
+        {
+            resourceOwnerSearch = search;
+            return this;
+        }
+
         public class Handler : IRequestHandler<GetDepartments, IEnumerable<QueryDepartment>>
         {
             private readonly ResourcesDbContext db;
+            private readonly IHttpClientFactory httpClientFactory;
+            private readonly IFusionProfileResolver profileResolver;
 
-            public Handler(ResourcesDbContext db)
-            {
+            public Handler(ResourcesDbContext db, IHttpClientFactory httpClientFactory,
+                IFusionProfileResolver profileResolver)
+            { 
                 this.db = db;
+                this.httpClientFactory = httpClientFactory;
+                this.profileResolver = profileResolver;
             }
 
             public async Task<IEnumerable<QueryDepartment>> Handle(GetDepartments request, CancellationToken cancellationToken)
             {
-                return await request.Execute(db.Departments).ToListAsync(cancellationToken);
+                var departments = await request.Execute(db.Departments).ToListAsync(cancellationToken);
+
+                if(request.shouldExpandResourceOwners)
+                {
+                    return await ExpandResourceOwners(departments, request.resourceOwnerSearch, cancellationToken);
+                }
+
+                return departments;
+            }
+
+            private async Task<List<QueryDepartment>> ExpandResourceOwners(List<QueryDepartment> departments, string? filter, CancellationToken cancellationToken)
+            {
+                var result = new List<QueryDepartment>();
+                var managedDepartments = departments.ToDictionary(dpt => dpt.DepartmentId);
+
+                var client = httpClientFactory.CreateClient("lineorg");
+
+                var uri = "/lineorg/persons?$filter=isresourceowner eq true";
+
+                if (!string.IsNullOrEmpty(filter))
+                    uri += $"&$search={filter}";
+
+                do
+                {
+                    var response = await client.GetAsync(uri, cancellationToken);
+                    response.EnsureSuccessStatusCode();
+
+                    var page = JsonSerializer.Deserialize<PaginatedResponse<ProfileWithDepartment>>(
+                        await response.Content.ReadAsStringAsync(cancellationToken),
+                        new JsonSerializerOptions { PropertyNameCaseInsensitive = true }
+                    );
+
+                    uri = page?.NextPage;
+                    var resourceOwners = page!.Value;
+
+                    foreach (var resourceOwner in resourceOwners)
+                    {
+                        if (!managedDepartments.ContainsKey(resourceOwner.FullDepartment)) continue;
+
+                        var department = managedDepartments[resourceOwner.FullDepartment];
+                        department.LineOrgResponsible = await profileResolver.ResolvePersonBasicProfileAsync(resourceOwner.AzureUniqueId);
```

Comment:
Iterating over an unkown number of items, making 1 or more external api call is not scalable. The resolver cache the profile, but the initial hit can be problematic. These resolvers support batch processing for this reason. 

What i usually do is to select all the ids that are relevant to resolve, then resolve all of them in one go before the iteration of internal model.